### ISSUE TITLE
Creating a rlua::String from a CString

### ISSUE THREAD

**Issue by althonos** [kyren/rlua#106]
Hi @kyren,

First of all thank you for this extremely useful library, I really enjoy working with it, since it manages to be close to the Lua API while providing a higher level of abstraction.

Yet I've been trying to create a function that returns a `rlua::String`, and couldn't find any way to do it: a `Vec<u8>` ends up being converted as a Table, and of course Rust `String` is not a suitable type to be converted since I'd want the string to contains binary data.

Given the `lua_pushstring` function accepts a `char*`, shouldn't it be possible to implement `ToLua` for `CString` types ? Otherwise, I can't see any way of creating a new `rlua::String` from the Rust code. 

Please do tell me if I'm missing something here ! :smile:

**Comment by kyren** [comment]
Yeah, that seems reasonable to me.  I think `ToLua` should probably be implemented for `CStr` and `CString`?  Should be a simple PR if you'd like to take a stab at it?
